src/plugin/AMD.ts

```diff
@@ -1,21 +1,13 @@
 import { Record, ModuleFactory } from '../Record';
 import { ModuleAMD } from '../Module';
 import { globalEnv, globalEval } from '../platform';
-import { Loader, LoaderConfig } from '../LoaderBase';
+import { Loader, LoaderPlugin } from '../Loader';
 
 /** AMD (asynchronous module definition) loader plugin. */
 
-export class AMD extends Loader {
+export const AMD = (loader: Loader): LoaderPlugin => {
```

Comment:
Why a function that returns a `LoaderPlugin` instead of a class that implements `LoaderPlugin`?

Reply:
I was also considering that. Might be better. I'll prototype that option too so we can compare the approaches.